### ISSUE TITLE
Using the VOC dataset for YOLO11 instance segmentation

### ISSUE THREAD

**Issue by wuchaoAAA** [ultralytics/ultralytics#19393]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

**### 1. Convert the mask PNG in the VOC dataset to YOLO format and generate a. txt file**
`import os
import cv2
import numpy as np
from PIL import Image

# 1️⃣ VOC 类别索引 (Pascal VOC 20 类 + 背景)
voc_classes = {
    0: "background", 1: "aeroplane", 2: "bicycle", 3: "bird", 4: "boat",
    5: "bottle", 6: "bus", 7: "car", 8: "cat", 9: "chair",
    10: "cow", 11: "diningtable", 12: "dog", 13: "horse", 14: "motorbike",
    15: "person", 16: "pottedplant", 17: "sheep", 18: "sofa", 19: "train",
    20: "tvmonitor"
}

# 2️⃣ 读取 VOC PNG 掩码
def convert_voc_mask_to_yolo(mask_path, output_txt_path):
    mask = np.array(Image.open(mask_path))  # 读取掩码 PNG
    h, w = mask.shape  # 获取图像大小

    # 过滤掉背景 (0) 和无效像素 (255)
    unique_labels = [label for label in np.unique(mask) if label not in [0, 255]]
    print(f"Processing: {mask_path} - Unique labels: {unique_labels}")

    with open(output_txt_path, 'w') as txt_file:
        for class_id in unique_labels:
            # 3️⃣ 提取轮廓
            binary_mask = np.uint8(mask == class_id) * 255  # 二值化
            contours, _ = cv2.findContours(binary_mask, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)

            for contour in contours:
                if len(contour) < 3:  # 需要至少 3 个点
                    continue

                # 4️⃣ 归一化坐标 (转换为 YOLO 格式)
                points = contour.reshape(-1, 2) / np.array([w, h])  # 归一化
                points_str = ' '.join(map(str, points.flatten()))

                # 5️⃣ 保存到 TXT (YOLO 格式: 类别索引 + 归一化坐标)
                txt_file.write(f"{class_id} {points_str}\n")

# 6️⃣ 批量转换所有 VOC 掩码
def batch_convert_voc_to_yolo(mask_dir, output_dir):
    if not os.path.exists(output_dir):
        os.makedirs(output_dir)

    for mask_file in os.listdir(mask_dir):
        if mask_file.endswith('.png'):
            mask_path = os.path.join(mask_dir, mask_file)
            output_txt_path = os.path.join(output_dir, mask_file.replace('.png', '.txt'))
            convert_voc_mask_to_yolo(mask_path, output_txt_path)

# 执行转换 (修改为你的路径)
batch_convert_voc_to_yolo('./VOC2012/SegmentationClass/', './1_instance_segment/datasets/VOC/labels_voc/')
`

### **2. Generate the following txt file**

![Image](https://github.com/user-attachments/assets/7aaf4df5-16dd-472a-81f0-27da45f6d5ce)

**### 3. voc-seg.yaml**

![Image](https://github.com/user-attachments/assets/175e67df-47e5-4963-a064-71b9e15df56b)

![Image](https://github.com/user-attachments/assets/ed5a2586-0a0f-4277-85f9-a1365581010c)
### 
**4.run code about train_yolo**
from ultralytics import YOLO
def train_model():
    # Load a model
    model = YOLO("./runs/segment/train/weights/best.pt")  #经过CARpart数据集训练后生成的分割模型
    results = model.train(data="VOC_seg.yaml", epochs=100, imgsz=640)
    # 保存模型
    model.save("./models/trained_coco_carparts_VOC_yolov11x.pt")  # 保存训练后的模型

if __name__ == '__main__':
    train_model()

**_According to the above procedure, if you want to use the png mask image of the VOC dataset as a yolo trainable txt file for instance segmentation, why does it show that the label cannot be found?_**

**Comment by UltralyticsAssistant** [comment]
👋 Hello @wuchaoAAA, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, many of which may address your question.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us understand and debug the issue. For your case, this would include specific details about the input dataset structure, the environment in which you're running your code, and the exact error or output you're encountering.

If this is a custom training ❓ Question, please provide as much information as possible about the problem you're experiencing, including:
- Details about your dataset (e.g., number of images, any modifications made to the mask files).
- Your full `VOC_seg.yaml` file configuration.
- Complete training logs, as they contain clues about any misconfigured paths, incompatible settings, or unexpected results.

Make sure you're following our [Best Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) and have upgraded to the latest version of the `ultralytics` package with all necessary requirements in a [**Python>=3.8**](https://www.python.org/) environment:

```bash
pip install -U ultralytics
```

## Community and Resources
Join the Ultralytics community to explore insights, share results, or seek additional guidance:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat.
- [Discourse](https://community.ultralytics.com/) for detailed technical discussions.
- [Subreddit](https://reddit.com/r/Ultralytics) to connect with the community.

## Supported Environments
YOLO has been verified to work in the following environments. You might consider verifying your code within one of these setups:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/).
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/).
- **Docker Image** for consistent environments: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>.

## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing. This ensures all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) are functioning correctly.

⚠️ Please note that this is an automated response, and an Ultralytics engineer will assist you further as soon as possible. 😊

**Comment by Y-T-G** [comment]
The dataset needs to be structured as shown in the docs.

https://docs.ultralytics.com/datasets/detect/#ultralytics-yolo-format
